### PR TITLE
Change Durable Objects "Hello, World!" template to use RPC methods

### PR DESCRIPTION
**Pull request by joshthoward**
## What this PR solves / how to test

Replaces https://github.com/cloudflare/workers-sdk/pull/5624.

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: Covered by existing tests
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because:

### CODE REVIEW

File: packages/create-cloudflare/templates/hello-world-durable-object/js/src/index.js [hunk 1]
```diff
@@ -16,25 +18,27 @@
  */
 
 /** A Durable Object's behavior is defined in an exported Javascript class */
-export class MyDurableObject {
+export class MyDurableObjec extends DurableObject {
 	/**
 	 * The constructor is invoked once upon creation of the Durable Object, i.e. the first call to
-	 * 	`DurableObjectStub::get` for a given identifier
+	 * 	`DurableObjectStub::get` for a given identifier (no-op constructors can be omitted)
```
Comment (penalosa) [inline packages/create-cloudflare/templates/hello-world-durable-object/js/src/index.js:24]: We say no op constructors can be omitted, but then include one in the example?

**Review by RamIdeas — APPROVED** [review]

File: packages/create-cloudflare/templates/hello-world-durable-object/ts/package.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
 		"start": "wrangler dev"
 	},
 	"devDependencies": {
-		"@cloudflare/workers-types": "^4.20231218.0",
```
Comment (jculvey) [inline packages/create-cloudflare/templates/hello-world-durable-object/ts/package.json:11]: C3 already installs the latest version of `@cloudflare/workers-types` as part of the template execution process, so this line shouldn't be needed.

**Review by jculvey — APPROVED** [review]
Left a comment but other than that lgtm

File: packages/create-cloudflare/templates/hello-world-durable-object/js/src/index.js [hunk 3]
```diff
@@ -16,25 +18,27 @@
  */
 
 /** A Durable Object's behavior is defined in an exported Javascript class */
-export class MyDurableObject {
+export class MyDurableObjec extends DurableObject {
```
Comment (jbwcloudflare) [inline packages/create-cloudflare/templates/hello-world-durable-object/js/src/index.js:21]: typo here `MyDurableObjec` is missing the "t" at the end

File: packages/create-cloudflare/templates/hello-world-durable-object/js/src/index.js (hunk 1 above)
Comment (joshthoward) [reply packages/create-cloudflare/templates/hello-world-durable-object/js/src/index.js:24]: Yeah. I think that it's helpful to include by default but note that it isn't explicitly required.

File: .changeset/serious-socks-grab.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"create-cloudflare": minor
+---
+
+Change Durable Objects "Hello, World!" template to use RPC methods
```
Comment (petebacondarwin) [inline .changeset/serious-socks-grab.md:5]:
```suggestion
feat: change Durable Objects "Hello, World!" template to use RPC methods
```

**Review by petebacondarwin — APPROVED** [review]
Minor nit on the changeset - please can you update it to follow our required guidelines as proposed.

Also, do we actually want people to stop using the `fetch` approach? Should we consider "adding" the JS RPC option to these rather than replacing the current fetch entry-point?

**Pull request merged**